exercises/simple-linked-list/linked_list_test.go

```diff
@@ -0,0 +1,191 @@
+// func (*List) Reverse() *List
+
+package linkedlist
+
+import (
+	"reflect"
+	"testing"
+)
+
+var array1To10 = []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
+
+func TestEmptyListHasSizeZero(t *testing.T) {
+	list := New([]int{})
+	size := list.Size()
+	if size != 0 {
+		t.Errorf("Size of empty list: %d, expected: %d", size, 0)
+	}
+}
+
+func TestSingletonListHasSizeOne(t *testing.T) {
+	list := New([]int{1})
+	size := list.Size()
+	if size != 1 {
+		t.Errorf("Size of singleton list: %d, expected: %d", size, 1)
+	}
+}
+
+func TestNonEmptyListHasCorrectSize(t *testing.T) {
+	list := New([]int{1, 2, 3})
+	size := list.Size()
+	if size != 3 {
+		t.Errorf("Size of list from [1, 2, 3]: %d, expected: %d", size, 3)
+	}
+}
+
+func TestEmptyListToEmptyArray(t *testing.T) {
+	list := New([]int{})
+	array := list.Array()
+	if len(array) != 0 {
+		t.Errorf("Test empty list to array: %v, want empty array", array)
+	}
+}
+
+func TestNonEmptyListToArray(t *testing.T) {
+	expected := []int{1, 2, 3}
+	list := New(expected)
+	array := list.Array()
+	if !reflect.DeepEqual(array, expected) {
+		t.Errorf("Test non empty list to array: %v, want %v", array, expected)
+	}
+}
+
+func TestPopFromEmptyList(t *testing.T) {
+	list := New([]int{})
+	err := list.Pop()
+	if err == nil {
+		t.Errorf("Pop from empty list: expected error but there was not")
+	}
+}
+
+func TestPopFromNonEmptyList(t *testing.T) {
+	list := New([]int{1, 2, 3})
+	err := list.Pop()
+	if err != nil {
+		t.Errorf("Pop from non empty list: unexpected error %v", err)
+	}
+	actual := list.Array()
+	expected := []int{1, 2}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Pop from non empty list: %v, want %v", actual, expected)
+	}
+}
+
+func TestPushToEmptyList(t *testing.T) {
+	list := New([]int{})
+	list.Push(1)
+	actual := list.Array()
+	expected := []int{1}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Push to empty list: %v, want %v", actual, expected)
+	}
+}
+
+func TestPushToNonEmptyList(t *testing.T) {
+	list := New([]int{1, 2, 3})
+	list.Push(4)
+	actual := list.Array()
+	expected := []int{1, 2, 3, 4}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Push to non empty list: %v, want %v", actual, expected)
+	}
+}
+
+func TestPushAndPop(t *testing.T) {
+	list := New([]int{1, 2, 3})
+	list.Pop()
+	list.Push(4)
+	list.Push(5)
+	list.Pop()
+	list.Push(6)
+	actual := list.Array()
+	expected := []int{1, 2, 4, 6}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Test push and pop: %v, want %v", actual, expected)
+	}
+}
+
+func TestReverseEmptyList(t *testing.T) {
+	list := New([]int{})
+	reversed := list.Reverse().Array()
+	if len(reversed) != 0 {
+		t.Errorf("Reverse empty list: %v, want empty list", reversed)
+	}
+}
+
+func TestReverseNonEmptyList(t *testing.T) {
+	list := New([]int{1, 2, 3})
+	actual := list.Reverse().Array()
+	expected := []int{3, 2, 1}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Reverse non empty list: %v, want %v", actual, expected)
```

Comment:
Since there is one case, it doesn't matter here but we prefer `Fatalf`. Could you please use it ?